pkg/storage/common.go

```diff
@@ -76,21 +123,32 @@ func ValidateManifest(imgStore ImageStore, repo, reference, mediaType string, bo
 	switch mediaType {
 	case ispec.MediaTypeImageManifest:
 		var manifest ispec.Manifest
+
+		// validate manifest
+		if err := validateOCIv1ImageManifest(body, log); err != nil {
+			log.Error().Err(err).Msg("OCIv1 image manifest schema validation failed")
+
+			return "", zerr.ErrBadManifest
+		}
+
 		if err := json.Unmarshal(body, &manifest); err != nil {
-			log.Error().Err(err).Msg("unable to unmarshal JSON")
+			log.Error().Err(err).Msg("unable to unmarshal image manifest")
 
 			return "", zerr.ErrBadManifest
 		}
 
+		// validate manifest config
 		if manifest.Config.MediaType == ispec.MediaTypeImageConfig {
-			digest, err := validateOCIManifest(imgStore, repo, reference, &manifest, log)
+			err := validateOCIv1ImageConfig(imgStore, repo, &manifest.Config, log)
 			if err != nil {
-				log.Error().Err(err).Msg("invalid oci image manifest")
+				log.Error().Str("digest", manifest.Config.Digest.String()).
+					Err(err).Msg("OCIv1 image config schema validation failed")
 
-				return digest, err
+				return "", err
 			}
 		}
 
+		// validate "subject"
 		if manifest.Subject != nil {
 			var m ispec.Descriptor
 			if err := json.Unmarshal(body, &m); err != nil {
```

Comment:
What exactly is this check doing? 
Manifest Subject is already type descriptor after the unmarshal?
This check verifies the manifest itself is a descriptior?
Do we need this?